Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -533,6 +533,11 @@ BlobType DirectoryBlobReader::GetBlobType() const
   return BlobType::DIRECTORY;
 }
 
+std::unique_ptr<BlobReader> DirectoryBlobReader::CopyReader() const
+{
+  return Create(CreateDisc(m_wrapped_volume->GetBlobReader().CopyReader()), {}, {});
```

Comment:
A DirectoryBlobReader can be created using one of two constructors, and only one of them sets a `m_wrapped_volume`. You need to handle both cases.

I also highly doubt that just passing {} for the remaining parameters works.